cli/src/pcluster/models/imagebuilder.py

```diff
@@ -187,6 +213,24 @@ def create(self, disable_rollback: bool = False):
             LOGGER.critical(e)
             raise ImageBuilderActionError(f"ImageBuilder stack creation failed.\n{e}")
 
+    def delete(self):
+        """Delete CFN Stack and associate resources and deregister the image."""
+        # validate image name
+        self._validate_image_name()
+
+        # check imagebuilder stack existence
+        if not AWSApi.instance().cfn.stack_exists(self.image_name):
```

Comment:
I think you can directly call deregister_image then delete_stack by catching exceptions, rather than checking if the stack exist, it permits to save one boto3 call and have the same result.

Reply:
good point, done